client/src/components/ProfileForm.js

```diff
@@ -0,0 +1,240 @@
+		setInputValues({ ...inputValues, [name]: value });
+	};
+
+	// Sends data to our back end server
+	const handleSubmit = async () => {
+		// Password validator
+		if (inputValues.password || inputValues.confirmPassword) {
+			// Checks if password matches
+			if (inputValues.password !== inputValues.confirmPassword) {
+				setSnackbar({
+					open: true,
+					severity: "error",
+					message: "Passwords do not match"
+				});
+				return;
+			}
+
+			// Checks if they are at least 6 characters
+			if (
+				inputValues.password.length < 6 ||
+				inputValues.confirmPassword.length < 6
+			) {
+				setSnackbar({
+					open: true,
+					severity: "error",
+					message: "Password must be at least 6 characters"
+				});
+				return;
+			}
+		}
+
+		// Removing confirmPassword before sending data
+		let copyOfInputValues = inputValues;
+		delete copyOfInputValues.confirmPassword;
+
+		if (type === "Register") {
+			// Checks for invalid email
+			if (!EmailValidator.validate(inputValues.email)) {
+				setSnackbar({
+					open: true,
+					severity: "error",
+					message: "Invalid email"
+				});
+				return;
+			}
+
+			try {
+				// Our data to send to the server
+				const res = await fetch(route, {
+					method: "POST",
+					body: JSON.stringify(copyOfInputValues),
+					headers: { "Content-Type": "application/json" }
+				});
+
+				// Redirect them to messenger page
+				if (res.status === 201) {
+					history.push("/messenger");
+				}
+
+				// In the event we get an error
+				const data = await res.json();
+				if (data.errors) {
+					if (data.errors.email) {
+						setSnackbar({
+							open: true,
+							severity: "error",
+							message: data.errors.email
+						});
+					}
+				}
+			} catch (err) {
+				console.log(err);
+			}
+		} else {
+			try {
+				// Our data to send to the server
+				const res = await fetch(route, {
+					method: "POST",
+					body: JSON.stringify(copyOfInputValues),
+					headers: { "Content-Type": "application/json" }
+				});
+
+				// Redirect them to messenger page
+				if (res.status === 201) {
+					window.location.reload();
+				}
+			} catch (err) {
+				console.log(err);
+			}
+		}
+	};
+
+	// Pressing enter submits our form
+	const handleKeyPress = (e) => {
+		if (e.which === 13) {
+			handleSubmit();
+		}
+	};
+
+	// Closes our snackbar
+	const handleClose = (event, reason) => {
+		if (reason === "clickaway") {
+			return;
+		}
+
+		setSnackbar({ ...snackbar, open: false });
+	};
+
+	const snackbars = (
+		<div>
+			<Snackbar
+				anchorOrigin={{
+					vertical: "top",
+					horizontal: "center"
+				}}
+				open={snackbar.open}
+				autoHideDuration={6000}
+				onClose={handleClose}
+			>
+				<Alert onClose={handleClose} severity={snackbar.severity}>
+					{snackbar.message}
+				</Alert>
+			</Snackbar>
+		</div>
+	);
+
+	// Error handling for password (must match and 6 chars minimum) for material-ui
+	let validPassword;
+	if (inputValues.password || inputValues.confirmPassword) {
```

Comment:
instead of putting this in the function itself, so it runs every single time something changes, it may be better practice to put it in a useEffect, every time inputValues changes

Reply:
I'll add this in my following PRs!